sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -36,18 +36,48 @@ namespace sycl {
 
 // Forward declarations
 class queue;
-namespace detail {
-class queue_impl;
-} // namespace detail
 
 namespace detail {
 
+enum class ExtendedMembersType: unsigned int {
+  HANDLER_KERNEL_BUNDLE = 0,
+};
+
+struct ExtendedMember {
+  ExtendedMembersType MType;
+  std::shared_ptr<void> MData;
+};
+
+static std::shared_ptr<std::vector<ExtendedMember>>
+convertToExtendedMembers(const std::shared_ptr<const void> &SPtr) {
+  return std::const_pointer_cast<std::vector<ExtendedMember>>(
+      std::static_pointer_cast<const std::vector<ExtendedMember>>(SPtr));
+}
+
 class stream_impl;
+class queue_impl;
+class kernel_bundle_impl;
+
+constexpr static unsigned int getVersionedCGType(unsigned int Type,
```

Comment:
I may be wrong, but `constexpr` used in a function implies `inline`, thus you don't need static.
```suggestion
constexpr unsigned int getVersionedCGType(unsigned int Type,
```